Library/Homebrew/brew.sh

```diff
@@ -1,8 +1,13 @@
 # Force UTF-8 to avoid encoding issues for users with broken locale settings.
-if [[ "$(locale charmap 2>/dev/null)" != "UTF-8" ]]
```

Comment:
What does `locale charmap` output for `zh_TW.UTF-8`?

Reply:
It shows UTF-8. However the warning will still be there, because LC_ALL was set to en_US.UTF-8. Here is the locale details.


```
=>locale
LANG=zh_TW.UTF-8
LANGUAGE=zh_TW:zh
LC_CTYPE="zh_TW.UTF-8"
LC_NUMERIC="zh_TW.UTF-8"
LC_TIME="zh_TW.UTF-8"
LC_COLLATE="zh_TW.UTF-8"
LC_MONETARY="zh_TW.UTF-8"
LC_MESSAGES="zh_TW.UTF-8"
LC_PAPER="zh_TW.UTF-8"
LC_NAME="zh_TW.UTF-8"
LC_ADDRESS="zh_TW.UTF-8"
LC_TELEPHONE="zh_TW.UTF-8"
LC_MEASUREMENT="zh_TW.UTF-8"
LC_IDENTIFICATION="zh_TW.UTF-8"
LC_ALL=zh_TW.UTF-8
```

Reply:
Was your locale set to `zh_TW.UTF-8` before running the `brew` bash script? If it was `zh_TW.UTF-8` originally, then the `export LC_ALL="en_US.UTF-8"` shouldn't have happened in the first place.

Reply:
As @apjanke said: if `locale charmap` outputs `UTF-8` it'll never set `LC_ALL`.